diff --git a/distribution/archives/build.gradle b/distribution/archives/build.gradle
index 14a7e56..01746d0 100644
--- a/distribution/archives/build.gradle
+++ b/distribution/archives/build.gradle
@@ -167,7 +167,7 @@ task buildOssNoJdkDarwinTar(type: Tar) {
 
 task buildLinuxTar(type: Tar) {
   configure(commonTarConfig)
-  archiveClassifier = 'linux-x86_64'
+  archiveClassifier = 'linux-s390x'
   with archiveFiles(modulesFiles(false, 'linux'), 'tar', 'linux', false, true)
 }
 
