diff --git a/bazel/repositories.bzl b/bazel/repositories.bzl
index 66d5d1aeb6..da6c99a6c9 100644
--- a/bazel/repositories.bzl
+++ b/bazel/repositories.bzl
@@ -689,7 +689,6 @@ def _com_github_gperftools_gperftools():
     http_archive(
         name = "com_github_gperftools_gperftools",
         build_file_content = BUILD_ALL_CONTENT,
-        patch_cmds = ["./autogen.sh"],
         **location
     )
 
@@ -751,5 +750,8 @@ def _is_arch(ctxt, arch):
 def _is_linux_ppc(ctxt):
     return _is_linux(ctxt) and _is_arch(ctxt, "ppc")
 
+def _is_linux_s390x(ctxt):
+    return _is_linux(ctxt) and _is_arch(ctxt, "s390x")
+
 def _is_linux_x86_64(ctxt):
     return _is_linux(ctxt) and _is_arch(ctxt, "x86_64")
