diff --git a/source/exe/BUILD b/source/exe/BUILD
index 43ff9f36fd..09a33a0f2a 100644
--- a/source/exe/BUILD
+++ b/source/exe/BUILD
@@ -43,6 +43,7 @@ envoy_cc_library(
     ] + select({
         "//bazel:windows_x86_64": envoy_windows_extensions(),
         "//bazel:linux_ppc": envoy_all_extensions(PPC_SKIP_TARGETS),
+	"//bazel:linux_s390x": envoy_all_extensions(),
         "//conditions:default": envoy_all_extensions(),
     }),
 )
