diff --git a/envoy.bazelrc b/envoy.bazelrc
index f1c01584..64b2bcfd 100644
--- a/envoy.bazelrc
+++ b/envoy.bazelrc
@@ -16,9 +16,9 @@ build --experimental_strict_action_env=true
 build --host_force_python=PY2
 build --action_env=BAZEL_LINKLIBS=-l%:libstdc++.a
 build --action_env=BAZEL_LINKOPTS=-lm
-build --host_javabase=@bazel_tools//tools/jdk:remote_jdk11
-build --javabase=@bazel_tools//tools/jdk:remote_jdk11
-build --enable_platform_specific_config
+build --host_javabase=@local_jdk//:jdk
+build --javabase=@local_jdk//:jdk
+#build --enable_platform_specific_config
 
 # Enable position independent code, this option is not supported on Windows and default on on macOS.
 build:linux --copt=-fPIC
